server/src/main/java/com/thoughtworks/go/server/service/JobInstanceService.java

```diff
@@ -261,11 +259,23 @@ public int totalCompletedJobsCountOn(String uuid) {
     }
 
     public JobInstancesModel completedJobsOnAgent(String uuid, JobHistoryColumns columnName, SortOrder order, Pagination pagination) {
-        List<JobInstance> jobInstances = jobInstanceDao.completedJobsOnAgent(uuid, columnName, order, pagination.getOffset(), pagination.getPageSize());
+        List<JobInstance> jobInstances = jobInstanceDao.completedJobsOnAgent(uuid,
+                columnName == JobHistoryColumns.duration ? JobHistoryColumns.completed : columnName,
+                order, pagination.getOffset(), pagination.getPageSize());
```

Comment:
We don't have a `duration` column in the sql resultset, is that why you're using the `completed` column? From what I see in the query in [JobInstance.xml](https://github.com/gocd/gocd/blob/master/server/src/main/resources/com/thoughtworks/go/server/dao/maps/JobInstance.xml#L290), the `completed` column is evaluated to `BuildStateTransition.stateChangeTime` for completed jobs.

So, in the sql, we end up sorting by when the job was completed, but that won't necessarily give us the correct order for the duration. I tested this out locally by running a few pipelines and then reducing the pagesize. The sort doesn't work properly when there are multiple pages.

This seems tricky to implement, but one way to do it would be to calculate the duration in the sql itself.

Reply:
Correct, I sorted by completed given that there is no duration column in table. I totally disregarded the fact that pagination happens in the sql instead of the code hence it doesn't work properly because of pagination. 

I'll do the calculation of duration in the sql query instead. I'll update the PR.